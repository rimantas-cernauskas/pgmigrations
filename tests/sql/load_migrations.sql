-- Test function pgmigrations.load_migrations
\i tests/sql/setup/init.sql
BEGIN;
SELECT plan(4);

-- setup mocks
\i tests/sql/setup/load_migrations_mocks.sql

SET ROLE __pgmigrations_testuser;

SELECT is(
     pgmigrations.load_migrations()
    ,TRUE
    ,'load_migrations() should return TRUE on successful load'
);
PREPARE my_query AS
    SELECT migration ,up ,down
    FROM pgmigrations.migrations_info
;

PREPARE my_expect AS
    SELECT * FROM (
        VALUES (
             '00'
            ,'CREATE TABLE foo( id SERIAL PRIMARY KEY );'
            ,'DROP TABLE foo;'
        ),(
             '02'
            ,'ALTER TABLE foo ADD COLUMN new_column TEXT NOT NULL;'
            ,NULL
        )
    )
    t(
         migration
        ,up
        ,down
    )
;

SELECT results_eq(
     'my_query'
    ,'my_expect'
    ,'should have migrations added'
);

SELECT is(
     pgmigrations.load_migrations()
    ,TRUE
    ,'load_migrations() re-run should return TRUE on successful load'
);

SELECT results_eq(
     'my_query'
    ,'my_expect'
    ,'should have same migrations list'
);
DEALLOCATE my_expect;
DEALLOCATE my_query;

SELECT * FROM finish();
ROLLBACK;
