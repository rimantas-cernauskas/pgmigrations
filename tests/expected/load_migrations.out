-- Test function pgmigrations.load_migrations
\i tests/sql/setup/init.sql
\set ECHO errors
1..4
NOTICE:  No "up.sql" script found, skipping migration "01"
ok 1 - load_migrations() should return TRUE on successful load
ok 2 - should have migrations added
NOTICE:  Migration "00" already loaded
NOTICE:  No "up.sql" script found, skipping migration "01"
NOTICE:  Migration "02" already loaded
ok 3 - load_migrations() re-run should return TRUE on successful load
ok 4 - should have same migrations list
